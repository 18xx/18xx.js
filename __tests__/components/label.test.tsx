import * as React from 'react';
import { ReactElement } from 'react';
import * as renderer from 'react-test-renderer';

import Label from '../../src/components/label';
import Tile from '../../src/components/tile';

import Hexagon from '../../src/hexagon';
import Point from '../../src/point';

describe('Label', () => {
  const hexagon: Hexagon = new Hexagon('east-west');

  describe('#toString()', () => {
    describe('when the point is not specified', () => {
      it('returns the svg for the element at the default position', () => {
        const label: ReactElement<Label> = (
          <Label hexagon={hexagon} labelStr='B' />
        );
        expect(renderer.create(label)).toMatchSnapshot();
      });
    });

    describe('when the point is specified', () => {
      describe('when the x point is less than the center', () => {
        it('returns start', () => {
          const label: ReactElement<Label> = (
            <Label hexagon={hexagon} labelStr='B' point={new Point(55, 0)} />
          );
          expect(renderer.create(label)).toMatchSnapshot();
        });
      });

      describe('when the x point is equal to the center', () => {
        it('returns middle', () => {
          const label: ReactElement<Label> = (
            <Label
            hexagon={hexagon}
            labelStr='B'
            point={new Point(Tile.WIDTH / 2, 0)} />
          );
          expect(renderer.create(label)).toMatchSnapshot();
        });
      });

      describe('when the x point is greater than the center', () => {
        it('returns end', () => {
          const label: ReactElement<Label> = (
            <Label hexagon={hexagon} labelStr='B' point={new Point(56, 0)} />
          );
          expect(renderer.create(label)).toMatchSnapshot();
        });
      });
    });
  });
});
