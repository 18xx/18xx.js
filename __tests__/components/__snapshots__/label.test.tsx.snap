// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label #toString() when the point is not specified returns the svg for the element at the default position 1`] = `
<text
  fill="black"
  fontSize={16}
  fontWeight="bold"
  textAnchor="end"
  x={109}
  y={46}
>
  B
</text>
`;

exports[`Label #toString() when the point is specified when the x point is equal to the center returns middle 1`] = `
<text
  fill="black"
  fontSize={16}
  fontWeight="bold"
  textAnchor="middle"
  x={55.42562584220407}
  y={0}
>
  B
</text>
`;

exports[`Label #toString() when the point is specified when the x point is greater than the center returns end 1`] = `
<text
  fill="black"
  fontSize={16}
  fontWeight="bold"
  textAnchor="end"
  x={56}
  y={0}
>
  B
</text>
`;

exports[`Label #toString() when the point is specified when the x point is less than the center returns start 1`] = `
<text
  fill="black"
  fontSize={16}
  fontWeight="bold"
  textAnchor="start"
  x={55}
  y={0}
>
  B
</text>
`;
